js/plugins/google-cloud/src/gcpOpenTelemetry.ts

```diff
@@ -211,6 +163,96 @@ export class GcpOpenTelemetry implements TelemetryConfig {
   }
 }
 
+/**
+ * Adjusts spans before exporting to GCP. In particular, redacts PII
+ * (input prompts and outputs), and adds a workaround attribute to
+ * error spans that marks them as error in GCP.
+ */
+class AdjustingTraceExporter implements SpanExporter {
+  constructor(private exporter: SpanExporter) {}
+
+  export(
+    spans: ReadableSpan[],
+    resultCallback: (result: ExportResult) => void
+  ): void {
+    this.exporter?.export(this.adjust(spans), resultCallback);
+  }
+
+  shutdown(): Promise<void> {
+    return this.exporter?.shutdown();
+  }
+
+  getExporter(): SpanExporter {
+    return this.exporter;
+  }
+
+  //forceFlush(): Promise<void> {
```

Comment:
remove?

Reply:
Whoops, added the actual implementation.